### PR TITLE
GCCDEV-4183:  Added `__serialize` and `__unserialize` method to `FractalDoctrinePaginatorAdapter` to allow easier caching.

### PR DESCRIPTION
**Pull request by davidxhill**

### CODE REVIEW

File: src/Pagination/FractalDoctrinePaginatorAdapter.php [hunk 1]
```diff
@@ -94,28 +81,30 @@ public function getPerPage(): int
...
-     */
-    public function getRouteGenerator()
+    public function __unserialize(array $data): void
     {
-        return $this->routeGenerator;
+        $this->total = $data['total'];
+        $this->url = $data['url'];
+        unset($data['total']);
+        unset($data['url']);
+        $this->request = RequestParams::fromArray($data);
```
Comment (rmarganti) [inline src/Pagination/FractalDoctrinePaginatorAdapter.php:108]: This has exposed a bug upstream in `giadc/json-api-request`.

If there is no sorting data, `RequestParams` uses an empty string here:

https://github.com/giadc/json-api-request/blob/master/src/Requests/RequestParams.php#L195

Exploding an empty string here…

https://github.com/giadc/json-api-request/blob/master/src/Requests/Sorting.php#L56

…creates the following for some mysterious PHP reason:

```php
[
    0 => '',
]
```

That causes bugs further down the line. We can avoid this issue by checking the `strlen()` of the string in the `Sorting` constructor.

**Review by rmarganti — CHANGES_REQUESTED** [review]
This PR is good, but the bug upstream in `json-api-request` needs addressed.

File: src/Pagination/FractalDoctrinePaginatorAdapter.php (hunk 1 above)
Comment (davidxhill) [reply src/Pagination/FractalDoctrinePaginatorAdapter.php:108]: completely forgot this pr existed. https://github.com/giadc/json-api-request/pull/8

Comment (davidxhill) [reply src/Pagination/FractalDoctrinePaginatorAdapter.php:108]: @rmarganti ^

**Pull request merged**